From d8360caf1e470336c63e02b4a61047dd8c6edee6 Mon Sep 17 00:00:00 2001
From: Nicolas Viéville <nicolas.vieville@uphf.fr>
Date: Sun, 28 Apr 2019 16:56:39 +0200
Subject: [PATCH] Add BUILD_FOR_RPM check in order to produce RPM package

Signed-off-by: Nicolas Viéville <nicolas.vieville@uphf.fr>
---
 Makefile | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/Makefile b/Makefile
index e56a479..8eeee9e 100644
--- a/Makefile
+++ b/Makefile
@@ -23,8 +23,12 @@ ifeq ($(strip $(DESTDIR)),)
   SUDO=
 else
   INSTALLBASE = $(DESTDIR)/usr/share/gnome-shell/extensions
+ifeq ($(BUILD_FOR_RPM),1)
+  SUDO=
+else
   SUDO=sudo
 endif
+endif
 
 ifdef VERSION
   VSTRING = _v$(VERSION)
@@ -96,13 +100,17 @@ install: remove build
 	$(call msg,$@,$(SUDO) $(INSTALLBASE)/$(INSTALLNAME))
 	$(Q) $(SUDO) mkdir -p $(INSTALLBASE)/$(INSTALLNAME)
 	$(Q) $(SUDO) cp $(VV) -r ./_build/* $(INSTALLBASE)/$(INSTALLNAME)/
+ifeq ($(strip $(BUILD_FOR_RPM)),)
 	$(Q) $(MAKE) -s reload
+endif
 	$(call msg,$@,OK)
 
 remove:
 	$(call msg,$@,$(SUDO) $(INSTALLBASE)/$(INSTALLNAME))
 	$(Q) $(SUDO) rm $(VV) -fr $(INSTALLBASE)/$(INSTALLNAME)
+ifeq ($(strip $(BUILD_FOR_RPM)),)
 	$(Q) $(MAKE) -s reload
+endif
 	$(call msg,$@,OK)
 
 reload:
-- 
2.20.1

